apps/api/src/cache/cache.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable, OnModuleDestroy } from '@nestjs/common'
+import { createClient, RedisClientType } from 'redis'
+import { User } from '@prisma/client'
+
+@Injectable()
+export class CacheService implements OnModuleDestroy {
+  private readonly redisClient: RedisClientType
+  private readonly userPrefix = 'user-'
+
+  constructor() {
+    this.redisClient = createClient({
+      url: 'redis://localhost:6379' // Replace with your Redis URL
```

Comment:
Read the redis url from the environment: https://github.com/keyshade-xyz/keyshade/blob/90b2e1472ffdc82cf7e0b2df57ae3659306ea806/.env.example#L4